File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/control/execution_control_response.py
from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from .execution_control_operation import AiSdkExecutionControlOperation
from .execution_control_state import AiSdkExecutionControlState


@dataclass(frozen=True)
class AiSdkExecutionControlResponse(AiSdkWireModel):
    execution_id: str
    operation: AiSdkExecutionControlOperation
    accepted: bool
    accepted_at_utc: str
    schema_version: int = AiSdkSchemaVersions.EXECUTION_CONTROL_RESPONSE
    state: AiSdkExecutionControlState | None = None
    correlation_id: str | None = None

File: implementations/node/sdk/src/contracts/control/execution-control-response.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkExecutionControlOperation } from "./execution-control-operation.js";
import type { AiSdkExecutionControlState } from "./execution-control-state.js";

export interface AiSdkExecutionControlResponse {
  readonly schemaVersion: typeof AI_SDK_SCHEMA_VERSIONS.executionControlResponse;
  readonly executionId: string;
  readonly operation: AiSdkExecutionControlOperation;
  readonly accepted: boolean;
  readonly state?: AiSdkExecutionControlState;
  readonly acceptedAtUtc: string;
  readonly correlationId?: string;
}
